src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -1043,3 +1042,19 @@
   [c :- CertOrCSR
    digest :- schema/Str]
   (SSLUtils/getFingerprint c digest))
+
+(schema/defn ^:always-validate prevent-duplicate :- (schema/maybe '(schema/Int))
+  "Given the list of serials to be added to the revoked list of Puppet's CA CRL
+   and the Puppet's CA CRL, extract the all the existing serial numbers
+   within the Puppet's CA CRL and group them into a list. Compare the given list
+   of serials against the newly extracted list for any shared serial numbers.
+   Those shared serial numbers will be removed. Return a list of serials to be
+   added to the revoked list without any numbers already existed in Puppet's CA CRL."
+  [serials :- '(schema/Int) ; must be a list of serials
+   cacrl :- X509CRL] ; must be an instance of X509CRL
+  (let [in? (fn [coll elem] (if (some #(= % elem) coll)
+              (do (println (str elem " existed. Will not be revoked")) true)
```

Comment:
```suggestion
              (do (println (str "Certificate with serial " elem " has already been revoked, will not revoke again.")) true)
```

Reply:
imo, this method doesn't need to be in this library. I wouldn't have its logging say "Will not be revoked" or "will not revoke again" because this function doesn't do any revoking itself.

Reply:
You mean the `prevent_duplicate` method? If so, I can happily move it over to the puppetserver repo.

Reply:
yeah, I was kinda thinking that too, since it's pretty specific to this particular use-case. In general, it's preferable to keep related code close together (same repo, or even same file) to make development easier, unless there's a more general use-case for it, in which case putting it somewhere shared makes sense.